sympy/operator/operator.py

```diff
@@ -0,0 +1,279 @@
+"""Provides the basic classes for operators."""
+
+from sympy.core.basic import Atom
+from sympy.core.compatibility import iterable
+from sympy.core.containers import Tuple
+from sympy.core.function import FunctionClass
+from sympy.core.expr import Expr
+from sympy.core.singleton import S
+from sympy.core.sympify import sympify
+
+class OpExpr(Expr):
+    """Basic class for every operator classes.  'Op' stands for 'Operator'.
+    """
+
+    _op_priority = 20 # This is some random big value
+
+    def __call__(self, *args, evaluate=True, **kwargs):
+        args = sympify(args)
+        return AppliedOp(self, args, evaluate=evaluate, **kwargs)
+
+    def _eval_operation(self, *args, **kwargs):
+        """Every subclasses of OpExpr should override this.
+        """
+        raise NotImplementedError()
```

Comment:
Can we give a message along with the error?

Reply:
Filled the message.